pkg/cache/internal/informers_map.go

```diff
@@ -434,12 +434,12 @@ func (w *gvkFixupWatcher) run() {
 		w.ch <- e
 	}
 	w.wg.Done()
+	close(w.ch)
```

Comment:
Should this be moved _above_ `w.wg.Done()` so that there's a guarantee that `w.ch` is closed after `w.Stop()` returns?

Reply:
I found watch.Filter which has the correct channel closing behaviour so I've re-implemented. See what you think.

Reply:
Nice! TIL about `watch.Filter`.